### PR TITLE
Fix Filebeat Zeek Weird Ingest Pipeline

### PR DESCRIPTION
**Pull request by xander33**
Some Zeek Weird logs do not contain IP addresses, causing the warning seen below:

**Logstash Output**
`[2020-01-28T15:49:35,993][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400, :action=>["index", {:_id=>nil, :_index=>"filebeat-zeek-2020.01.28", :routing=>nil, :_type=>"_doc", :pipeline=>"filebeat-7.5.2-zeek-weird-pipeline"}, #<LogStash::Event:0x3f1f2270>], :response=>{"index"=>{"_index"=>"filebeat-zeek-2020.01.28", "_type"=>"_doc", "_id"=>"r3PX7G8BxFIJZtUR_Ruu", "status"=>400, "error"=>{"type"=>"mapper_parsing_exception", "reason"=>"failed to parse field [destination.ip] of type [ip] in document with id 'r3PX7G8BxFIJZtUR_Ruu'. Preview of field's value: ''", "caused_by"=>{"type"=>"illegal_argument_exception", "reason"=>"'' is not an IP string literal."}}}}}`

**Sample from weird.log**
`{"ts":1580227259.342809,"name":"non_ip_packet_in_ethernet","notice":false,"peer":"ens3f1-4"}`

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
Hi @xander33, thanks for contributing! Could you also add a test file with Zeek Weird logs which doesn't contain IP address please? Similar to `x-pack/filebeat/module/zeek/files/test/files-json.log` and create an expected json file by running `MODULES_PATH=`pwd`/module GENERATE=1 INTEGRATION_TESTS=1 TESTING_FILEBEAT_MODULES=zeek nosetests -v -d -s tests/system/test_xpack_modules.py`.

File: x-pack/filebeat/module/zeek/weird/ingest/pipeline.json [hunk 1]
```diff
@@ -28,13 +28,15 @@
     {
       "set": {
         "field": "source.ip",
-        "value": "{{source.address}}"
+        "value": "{{source.address}}",
+        "if": "ctx.source.address != null"
```
Comment (andrewkroh) [inline x-pack/filebeat/module/zeek/weird/ingest/pipeline.json]: I recommend using the null-safe operator. https://www.elastic.co/guide/en/elasticsearch/painless/master/painless-operators-reference.html#null-safe-operator

```suggestion
        "if": "ctx?.source?.address != null"
```

File: x-pack/filebeat/module/zeek/weird/ingest/pipeline.json [hunk 2]
```diff
@@ -28,13 +28,15 @@
...
       }
     },
     {
       "set": {
         "field": "destination.ip",
-        "value": "{{destination.address}}"
+        "value": "{{destination.address}}",
+        "if": "ctx.destination.address != null"
```
Comment (andrewkroh) [inline x-pack/filebeat/module/zeek/weird/ingest/pipeline.json]:
```suggestion
        "if": "ctx?.destination?.address != null"
```

**Comment by xander33** [comment]
@andrewkroh good point.

@kaiyan-sheng How does that look?

**Comment by andrewkroh** [comment]
This needs a quick rebase to resolve the conflicts.

jenkins, test this

**Comment by kaiyan-sheng** [comment]
jenkins, test this please

**Comment by kaiyan-sheng** [comment]
Thanks @xander33 for contributing! I will merge this PR and cherrypick to 7.x branch after CI passed.

**Pull request merged**